File: python/800/boy_or_girl.py
import sys

input = sys.stdin.readline

def solve():
    n = input()
    only_org_letr = []
    for letter in n:
        if letter not in only_org_letr:
            only_org_letr.append(letter)
    indiv_number = len(only_org_letr) - 1   # as we have '/n' symbole from imput
    if indiv_number % 2 == 0:
        print("CHAT WITH HER!")
    else:
        print("IGNORE HIM!")


if __name__ == "__main__":
    solve()

File: C++/800/boy_or_girl.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    string w = "CHAT WITH HER!";
    string m = "IGNORE HIM!";

    string n;
    cin >> n;

    vector <char> list;

    for (char &letter : n ){
        if (find(list.begin(), list.end(), letter) == list.end()) {
            list.push_back(letter);
        }
    }

    if (list.size() % 2 == 0){
        cout << w << endl;
    }
    else {
        cout << m << endl;
    }
    
    return 0;
}